src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -520,14 +520,81 @@ void MPS::apply_matrix(const reg_t & qubits, const cmatrix_t &mat)
   case 1: 
     q_reg_[qubits[0]].apply_matrix(mat);
     break;
-  case 2:
-    apply_2_qubit_gate(qubits[0], qubits[1], su4, mat);
-    break;
+    //  case 2:
+    //    apply_2_qubit_gate(qubits[0], qubits[1], su4, mat);
+    //    break;
   default:
-    throw std::invalid_argument("currently support apply_matrix for 1 or 2 qubits only");
+    apply_multi_qubit_gate(qubits, mat);
   }
 }
 
+void MPS::apply_multi_qubit_gate(const reg_t &qubits,
+				 const cmatrix_t &mat) {
+  uint_t num_qubits = qubits.size();
+  // need to reverse qubits because that is the way they
+  // are defined in the Qiskit interface
+  reg_t reversed_qubits = qubits;
+  std::reverse(reversed_qubits.begin(), reversed_qubits.end()); 
+
+  bool ordered = true;
+  for (uint_t index=0; index < qubits.size()-1; index++) {
+    if (reversed_qubits[index]+1 != reversed_qubits[index+1]){
+      ordered = false;
+      break;
+    }
+  }
```

Comment:
just wondering, won't this be sufficient to know whether `reversed_qubits` are ordered or no?
```c++
bool ordered = num_qubits > 1 ? reversed_qubits[0]+1 == reversed_qubits[1] : true;
```

Reply:
No,  it is not sufficient, because I want to make sure **all qubits** are in order, not just the first 2.